### PR TITLE
Lisk Core 1.0 compatibility update merge branch - Closes #145

### LINKED ISSUE #145: Update API requests for Lisk Core 1.0.0
Update API requests to achieve compatibility with Lisk Core 1.0.0.

- [x] Update Api/accounts endpoints #287 https://github.com/LiskHQ/lisk-explorer/pull/288 #397 
- [x] Update Api/transactions endpoints #291 https://github.com/LiskHQ/lisk-explorer/pull/293 #398 
- [x] Update Api/blocks endpoints #290 https://github.com/LiskHQ/lisk-explorer/pull/298 #399 
- [x] Update Api/statistics endpoints #290 https://github.com/LiskHQ/lisk-explorer/issues/302 #400 
- [x] Update Api/search endpoints #303 https://github.com/LiskHQ/lisk-explorer/pull/305 #401 
- [x] Update Api/delegates endpoints #430 #431 
- [x] Update sockets/delegateMonitor.js #403 #437

### PR DESCRIPTION
**Pull request by MichalTuleja**
This PR collects all changes from #397 #398 #399 #400 #401 #402 #430 #464

The purpose of this ticket is to check if all merged changes from the tickets listed above are able to pass tests together. 

### Review checklist
- The PR solves #145
- All new code is covered with unit tests
- All new features are covered with e2e tests
- All new code follows best practices

### CODE REVIEW

File: lib/api/accounts.js [hunk 1]
```diff
@@ -35,48 +35,94 @@ module.exports = function (app) {
...
+			}, (err, response, body) => {
+				if (response.statusCode === 404 &&
+					typeof body === 'object' &&
+					body.message === 'Multisignature account not found'
+				) {
+					return cb(null, account);
+				} else if (err || response.statusCode !== 200) {
+					return cb(err || 'Response was unsuccessful');
+				} else if (body && Array.isArray(body.data) && body.data.length > 0) {
+					account.multisignatures = _.extend(account.multisignatures || []);
```
Comment (4miners) [inline lib/api/accounts.js]: Why we need to call `_.extend` here?

File: lib/api/accounts.js [hunk 2]
```diff
@@ -35,48 +35,94 @@ module.exports = function (app) {
...
 				json: true,
 			}, (err, response, body) => {
 				if (err || response.statusCode !== 200) {
 					return cb(err || 'Response was unsuccessful');
-				} else if (body.success === true) {
-					const account = body.account;
-					account.knowledge = knowledge.inAccount(account);
+				} else if (body && Array.isArray(body.data)) {
+					if (body.data.length > 0) {
+						account.multisignatures = _.extend(
```
Comment (4miners) [inline lib/api/accounts.js]: please add comment what is happening here

File: lib/api/accounts.js [hunk 3]
```diff
@@ -113,15 +157,15 @@ module.exports = function (app) {
...
 				json: true,
 			}, (err, response, body) => {
 				if (err || response.statusCode !== 200) {
 					return cb(err || 'Response was unsuccessful');
-				} else if (body.success === true) {
-					account.voters = (body.accounts !== undefined &&
-						body.accounts !== null && body.accounts.length > 0) ?
-						body.accounts : null;
+				} else if (body && Array.isArray(body.data)) {
+					account.voters = (body.data.voters !== undefined &&
```
Comment (4miners) [inline lib/api/accounts.js]: Simplify that check:
`account.voters = Array.isArray(body.data.voters) && body.data.voters.length > 0 ? body.data.voters : null;`

File: lib/api/blocks.js [hunk 4]
```diff
@@ -206,7 +252,7 @@ module.exports = function (app) {
 					if (err) {
 						return error({ success: false, error: err });
 					}
-					return success(result);
+					return success(_.extend({ success: true }, result));
```
Comment (4miners) [inline lib/api/blocks.js]: why `_.extend`?

File: lib/api/delegates.js [hunk 5]
```diff
@@ -28,34 +28,95 @@ module.exports = function (app) {
...
+				url: `${app.get('lisk address')}/delegates/?sort=rank:asc&limit=101`,
 				json: true,
 			}, (err, response, body) => {
 				if (err || response.statusCode !== 200) {
 					return cb(err || 'Response was unsuccessful');
-				} else if (body.success === true) {
-					body.delegates = parseDelegates(body.delegates);
-					return cb(null, body);
+				} else if (body.data) {
+					if (Array.isArray(body.data)) {
```
Comment (4miners) [inline lib/api/delegates.js]: merge that condition with above one

File: lib/api/delegates.js [hunk 6]
```diff
@@ -182,37 +240,39 @@ module.exports = function (app) {
...
+				url: `${app.get('lisk address')}/delegates?publicKey=${tx.senderPublicKey}`,
 				json: true,
 			}, (err, response, body) => {
 				if (err || response.statusCode !== 200) {
 					tx.delegate = null;
 					return cb();
-				} else if (body.success === true) {
-					tx.delegate = body.delegate;
-					return cb();
+				} else if (body.data) {
```
Comment (4miners) [inline lib/api/delegates.js]: add check for array

File: lib/api/delegates.js [hunk 7]
```diff
@@ -240,15 +300,16 @@ module.exports = function (app) {
...
 		request.get({
-			url: `${app.get('lisk address')}/api/delegates/getNextForgers?limit=101`,
+			url: `${app.get('lisk address')}/delegates/forgers?limit=100`, // TODO: Change to 100 after lisk-core update
 			json: true,
 		}, (err, response, body) => {
 			if (err || response.statusCode !== 200) {
 				return error({ success: false, error: (err || 'Response was unsuccessful') });
-			} else if (body.success === true) {
-				return success({ success: true, delegates: body.delegates });
+			} else if (Array.isArray(body.data)) {
```
Comment (4miners) [inline lib/api/delegates.js]: body &&

File: lib/api/delegates.js [hunk 8]
```diff
@@ -307,27 +368,28 @@ module.exports = function (app) {
...
 			request.get({
-				url: `${app.get('lisk address')}/api/blocks?&orderBy=height:desc&limit=1`,
+				url: `${app.get('lisk address')}/blocks?&sort=height:desc&limit=1`,
 				json: true,
 			}, (err, response, body) => {
 				if (err || response.statusCode !== 200) {
 					return cb(err || 'Response was unsuccessful');
-				} else if (body.success === true && _.size(body.blocks) === 1) {
-					return cb(null, body.blocks[0]);
+				} else if (_.size(body.data) === 1) {
```
Comment (4miners) [inline lib/api/delegates.js]: body &&

File: lib/api/delegates.js [hunk 9]
```diff
@@ -307,27 +368,28 @@ module.exports = function (app) {
...
 			request.get({
-				url: `${app.get('lisk address')}/api/delegates/get?publicKey=${block.generatorPublicKey}`,
+				url: `${app.get('lisk address')}/delegates?publicKey=${block.generatorPublicKey}`,
 				json: true,
 			}, (err, response, body) => {
 				if (err || response.statusCode !== 200) {
 					return cb(err || 'Response was unsuccessful');
-				} else if (body.success === true) {
-					block.delegate = body.delegate;
+				} else if (body.data) {
```
Comment (4miners) [inline lib/api/delegates.js]: body && Array.isArray(body.data)

File: lib/api/delegates.js [hunk 10]
```diff
@@ -362,14 +424,14 @@ module.exports = function (app) {
...
-			url: `${app.get('lisk address')}/api/blocks?orderBy=height:desc&generatorPublicKey=${params.publicKey}&limit=${params.limit}`,
+			url: `${app.get('lisk address')}/blocks?sort=height:desc&generatorPublicKey=${params.publicKey}&limit=${params.limit}`,
 			json: true,
 		}, (err, response, body) => {
 			if (err || response.statusCode !== 200) {
 				return error({ success: false, error: err });
 			}
-			body.blocks = _.isArray(body.blocks) ? body.blocks : [];
-			return success({ success: true, blocks: body.blocks });
+			body.blocks = _.isArray(body.data) ? body.data : [];
```
Comment (4miners) [inline lib/api/delegates.js]: use Array.isArray instead lodash

File: lib/api/statistics.js [hunk 11]
```diff
@@ -126,13 +158,14 @@ module.exports = function (app) {
...
-		}, (err, resp, body) => {
-			if (err || resp.statusCode !== 200) {
+		}, (err, res, body) => {
+			if (err || res.statusCode !== 200) {
 				return error({ success: false, error: (err || 'Response was unsuccessful') });
 			}
 
-			if (_.size(body.blocks) > 0) {
-				return success({ success: true, block: body.blocks[0] });
+			if (_.size(body.data) > 0) {
```
Comment (4miners) [inline lib/api/statistics.js]: body &&

File: lib/api/accounts.js [hunk 12]
```diff
@@ -35,48 +35,94 @@ module.exports = function (app) {
...
-					const account = body.account;
-					account.knowledge = knowledge.inAccount(account);
-					return cb(null, account);
+				} else if (body && Array.isArray(body.data)) {
+					if (body.data.length > 0) {
+						const account = body.data[0];
+						account.knowledge = knowledge.inAccount(account);
+						return cb(null, account);
+					}
+					return cb(null, []);
```
[the remark this answers is not in the document]
Comment (MichalTuleja) [reply lib/api/accounts.js]: This part is consistent with the new API, where you always get an array.
https://app.swaggerhub.com/apis/LiskHQ/Lisk/1.0.25#/Accounts/getAccounts

File: lib/api/accounts.js (hunk 1 above)
Comment (MichalTuleja) [reply lib/api/accounts.js]: It can be removed.

File: lib/api/accounts.js (hunk 2 above)
Comment (MichalTuleja) [reply lib/api/accounts.js]: Underscore _.extend() merges two objects into one. In this example, we gather all members and combine them with the present data.

File: lib/api/blocks.js (hunk 4 above)
Comment (MichalTuleja) [reply lib/api/blocks.js]: Again, we combine the status with the actual result.

File: lib/api/accounts.js [hunk 13]
```diff
@@ -35,48 +35,94 @@ module.exports = function (app) {
...
+			}, (err, response, body) => {
+				if (err || response.statusCode !== 200) {
+					return cb(err || 'Response was unsuccessful');
+				} else if (body && Array.isArray(body.data)) {
+					if (body.data.length > 0) {
+						const account = body.data[0];
+						account.knowledge = knowledge.inAccount(account);
+						return cb(null, account);
+					}
+					return cb(null, []);
```
[the remark this answers is not in the document]
Comment (MichalTuleja) [reply lib/api/accounts.js]: Please look the answer above

File: lib/api/accounts.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (4miners) [reply lib/api/accounts.js]: Core API doesn't matter, I meant: `return cb(null, account);` vs `return cb(null, []);` - if success we return object, if fail - empty array

File: lib/api/accounts.js (hunk 2 above)
Comment (4miners) [reply lib/api/accounts.js]: Yep, I know, but would be nice to explain (as a comment in code) why we merge `account.multisignatures` with `members`

**Pull request merged**